ui/apps/platform/src/Components/PatternFly/IconText/IconText.tsx

```diff
@@ -0,0 +1,27 @@
+import React, { ReactElement } from 'react';
+
+export type IconTextProps = {
+    Icon: ReactElement;
+    text: string;
+    isTextOnly?: boolean;
+};
+
+function IconText({ Icon, text, isTextOnly }: IconTextProps): ReactElement {
+    if (isTextOnly) {
+        // Wrap text in fragment so function returns ReactElement,
+        // because return ReactNode can cause error in caller:
+        // TS2786 IconText cannot be used as a JSX component.
```

Comment:
Good news, recently merged in TypeScript and shouldn't be necessary in the near future: https://www.github.com/microsoft/TypeScript/pull/51328